Two Pointers/three_sum.py

```
@@ -0,0 +1,74 @@
+
+def three_sum_2(nums):
+    result = []
+    nums.sort()
+
+    for i, a in enumerate(nums):
+        if i > 0 and a == nums[i - 1]:     # To skip the duplicate values of 'a'
+            continue
+
+        l, r = i + 1, len(nums) - 1
+        while l < r:
+            sum = a + nums[l] + nums[r]
```

Ruff (0.15.20)

[error] Variable `sum` is shadowing a Python builtin

(A001)